plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/util/ServerRSyncUtil.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.core.util;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.wst.server.core.IServer;
+import org.jboss.tools.openshift.common.core.utils.StringUtils;
+import org.jboss.tools.openshift.core.server.OpenShiftServerUtils;
+import org.jboss.tools.openshift.core.server.RSync;
+import org.jboss.tools.openshift.internal.core.OpenShiftCoreActivator;
+import org.jboss.tools.openshift.internal.core.preferences.OCBinary;
+
+import com.openshift.restclient.model.IService;
+
+public class ServerRSyncUtil {
```

Comment:
Seems pretty difficult... it'd require having actual projects in the workspace.  It all comes back to getDeployProject where a number of settings are stored..., so unless I can mock the entire workspace for the call to locate the project, I don't think any of this will work.  

OpenShiftServerUtils is basically one huge blob of untestable static methods.